### PR TITLE
task(admin-panel): Hookup account delete

### PR DESCRIPTION
**Pull request by dschom**
## Because

- We want to delete accounts from the admin panel

## This pull request

- Adds cloud task emulator pm2 service
- Creates cloud-task nx lib for interacting with cloud tasks
- Creates PageDeleteAccount in admin panel
- Adds ability to delete accounts to admin server's account resolver
- Fixes broken admin panel configuration
- Fixes broken admin server configuration

## Issue that this pull request solves

Closes:FXA-8862

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)

image

## Other information (Optional)

In order to not disrupt the auth server. **Only minimal changes were made in auth server**. I've filed a [follow up](https://mozilla-hub.atlassian.net/browse/FXA-9270) for using the 'cloud-tasks' nx lib in auth server pending approval of this PR.

I've also added the cloud task emulator in this PR, which allowed me to test locally. The only part that needed a work around was relaxing the oidc auth requirement in auth server, when the emulator was being used instead of an actual GCP service.

In order to test with the emulator, the default configs should be sufficient. To manually test 

- run `yarn start`
- create a new account
- navigate to the admin panel's delete account section
- type in the account email (or account uid), and press delete account 

The expected behavior is that the account show up as in the results area as 'pending', and then a few seconds later switch to 'task completed'.

### CODE REVIEW

File: packages/fxa-admin-panel/src/components/PageAccountDelete/index.tsx [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        setResults([]);
+        return;
+      }
+
+      const result = await deleteAccountMutation({
+        variables: { locators: list },
+      });
+      setResults(result.data?.deleteAccounts || []);
+    } catch (error) {
+      console.error(error);
```
Comment (vbudhram) [inline packages/fxa-admin-panel/src/components/PageAccountDelete/index.tsx:94]: Stray console

File: packages/fxa-admin-panel/src/components/PageAccountDelete/index.tsx [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+  return (
+    <>
+      <h2 className="header-page">Delete Accounts</h2>
+      <p>
+        This page allows you to delete user accounts.{' '}
+        <b>Be careful! Data will be lost through this action!</b>
+      </p>
+      <br />
+      <p>
+        Provide a list of up to 1000 emails or UIDs separated by either a coma
```
Comment (vbudhram) [inline packages/fxa-admin-panel/src/components/PageAccountDelete/index.tsx:109]: Was the 1000 based on the delete script? It does seem kind of high imo

File: packages/fxa-shared/guards/AdminPanelGuard.ts [hunk 3]
```diff
@@ -96,6 +98,10 @@ const defaultAdminPanelPermissions: Permissions = {
     name: 'Account History',
     level: PermissionLevel.Support,
   },
+  [AdminPanelFeature.AccountDelete]: {
+    name: 'Delete Account By Email/UID',
+    level: PermissionLevel.Admin,
```
Comment (vbudhram) [inline packages/fxa-shared/guards/AdminPanelGuard.ts:103]: 👍🏽

File: packages/fxa-auth-server/lib/routes/cloud-tasks.ts [hunk 4]
```diff
@@ -48,7 +48,7 @@ export const cloudTaskRoutes = (log: AuthLogger, config: ConfigType) => {
       path: accountDeleteCloudTaskPath,
       options: {
         auth: {
-          mode: 'required',
+          mode: config.cloudTasks.useLocalEmulator ? 'try' : 'required',
```
Comment (vbudhram) [inline packages/fxa-auth-server/lib/routes/cloud-tasks.ts:51]: Does the auth-server need anything else updated to use shared library?

**Review by vbudhram — APPROVED** [review]
@dschom Tested this locally and works well. Thanks for adding the cloud task emulator, just a few questions/nits otherwise!

**Comment by chenba** [comment]
> adding the cloud task emulator

👀

File: packages/fxa-admin-panel/src/components/PageAccountDelete/index.tsx (hunk 2 above)
Comment (dschom) [reply packages/fxa-admin-panel/src/components/PageAccountDelete/index.tsx:109]: This is just a ball park estimate that could be quite off. There was no actual upper bound. I didn't want to totally clobber the server though, so put some limit in place.

File: packages/fxa-auth-server/config/index.ts [hunk 5]
```diff
@@ -2024,51 +2025,7 @@ const convictConf = convict({
       format: Boolean,
     },
   },
-
-  cloudTasks: {
```
Comment (dschom) [inline packages/fxa-auth-server/config/index.ts:2028]: Config before refactor

File: libs/shared/cloud-tasks/src/lib/cloud-tasks.factories.ts [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+import { CloudTasksConfig } from './cloud-tasks.types';
+import { credentials } from '@grpc/grpc-js';
+
+/**
+ * Produces configuration required for running cloud tasks
+ * @param prefix - An optional prefix to target different environment variable sets
+ * @returns
+ */
+export function CloudTasksConvictConfigFactory() {
+  return {
```
Comment (dschom) [inline libs/shared/cloud-tasks/src/lib/cloud-tasks.factories.ts:15]: Config after refactor

File: libs/shared/cloud-tasks/src/lib/cloud-tasks.factories.ts [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+    const client = new CloudTasksClient({
+      port: 8123,
+      servicePath: 'localhost',
+      sslCreds: credentials.createInsecure(),
+    });
+
+    return client;
+  }
+
+  const cloudTasksClient = new CloudTasksClient({
```
Comment (dschom) [inline libs/shared/cloud-tasks/src/lib/cloud-tasks.factories.ts:86]: Here's where the client is configured...

**Pull request merged**